### ISSUE TITLE
Silent bug in pretrained feature usage

### ISSUE THREAD

**Issue by paulsonak** [ATOMScience-org/AMPL#237]
I am using AMPL 1.6.0
1. I created a dataset and featurized it for later use (rdkit_raw)
2. I realized I curated the dataset incorrectly and re-created it, saving it with the same name
3. I forgot to update the featurized file in the scaled descriptor folder
4. The featurized dataset is matched by file name; it did not contain an equal number of rows as the real dataset
    a. it contained more compounds than the dataset
    b. It might have contained the real compounds plus additional rows, or
    c. It might have not contained all the real compounds - only some of them, missing some and having others instead
5. The model training (basic NN) process basically failed. Train r2 was -0.000 for every epoch. But, no error and the training completed successfully.

We should 
1. include a check if the real data is contained in the featurized data but allow extra compounds to exist in the featurized file
2. throw an error if there are unfeaturized compounds in a merged dataset file or throw a warning and re-featurize during the model process
4. think about replacing the file-name match with something more robust

**Comment by paulsonak** [comment]
I create a branch `prefeat_smiles_check` and [checked in an exception](https://github.com/ATOMScience-org/AMPL/commit/aa7952691f08c64cc22c22eb4df548c2b3ca6921).

I put the exception in `model_datasets.py` inside `load_featurized_data` but `load_featurized_data` is called by [`get_featurized_data`](https://github.com/ATOMScience-org/AMPL/blob/84655a2ef0cf077a55452c859519abba1452806b/atomsci/ddm/pipeline/model_datasets.py#L349) which "Does whatever is necessary to prepare a featurized dataset."

So basically our previous philosophy was to fix an error like this or acommodate the user's mistake by creating the featurized dataset, but that would involve overwriting a user's featurized dset file. @stewarthe6 @mcloughlin2 thoughts?
